services/director-v2/src/simcore_service_director_v2/modules/comp_scheduler/_task.py

```diff
@@ -0,0 +1,49 @@
+import datetime
+import logging
+from collections.abc import Callable, Coroutine
+from typing import Any, Final
+
+from fastapi import FastAPI
+from servicelib.background_task import start_periodic_task, stop_periodic_task
+from servicelib.logging_utils import log_context
+from servicelib.redis import RedisClientsManager
+from servicelib.redis_utils import exclusive
+from settings_library.redis import RedisDatabase
+
+from . import _scheduler_factory
+
+_logger = logging.getLogger(__name__)
+
+_COMPUTATIONAL_SCHEDULER_INTERVAL: Final[datetime.timedelta] = datetime.timedelta(
+    seconds=5
+)
+_TASK_NAME: Final[str] = "computational services scheduler"
+
+
+def on_app_startup(app: FastAPI) -> Callable[[], Coroutine[Any, Any, None]]:
```

Comment:
THOUGHT: would it make sense to abstract these out as parameterized functions and just reduce all these to  some setup function in a module within `servicelib.fastapi`. 

I asked to chatty for some name suggestions that stress what in my opinion are the important features :
Given your requirements, here are some name suggestions that highlight the periodic, background nature of the task and its single-execution guarantee across replicas:

1. **`singleton_background_task`** – Emphasizes a single instance of the background task running at a time.
2. **`exclusive_periodic_task`** – Highlights the periodic nature and exclusivity across replicas.
3. **`unique_background_runner`** – Conveys a single, constantly running background task.
4. **`replica_safe_task`** – Captures that the task is safe to run across replicas but only one at a time.
5. **`distributed_periodic_worker`** – Suggests distributed coordination with periodic execution.
6. **`singleton_scheduled_task`** – Emphasizes a singleton pattern for the scheduled task.

Each of these names conveys the unique, constantly running, and singleton-enforced execution behavior across replicas. Let me know if you'd like further customization based on a particular style!

Reply:
Note that we can grow a good collection of these background tasks with different policy requirements e.g. some might be ok to run them in all replicas, some might be a one time run and die etc ... (cron jobs like?)

Reply:
hmm. well we can discuss that. I am not super found of these names, not sure I understand what they bring here. I prefer composition over slightly unclear names. but yeah, anyway this will further change in the next PRs.
This one is for refactoring.